mesonbuild/linkers/detect.py

```diff
@@ -71,7 +72,10 @@ def guess_win_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
         elif not invoked_directly:
             return linkers.ClangClDynamicLinker(
                 for_machine, override, exelist=compiler, prefix=comp_class.LINKER_PREFIX,
-                version=search_version(o), direct=False, machine=None)
+                version=search_version(o), direct=False, machine=None,
+                rsp_syntax=RSPFileSyntax.MSVC
+                if comp_class.get_argument_syntax() == 'msvc'
+                else RSPFileSyntax.GCC)
```

Comment:
I find this line wrapping hard to read.

Reply:
The wrapping style is required by linter. I didn't have much choice.

Reply:
I'm pretty sure our linter configuration doesn't include rules such as requiring that the `x if y else z` construct be split across three lines.

In general I'm fairly skeptical at the idea of linters that "require" you to do things a certain way. The job of a linter is usually to detect aberrant patterns and tell you not to use them, not to be prescriptive about what you do to solve it. That is e.g. why meson does not (and never will) use `black`

Reply:
> I'm pretty sure our linter configuration doesn't include rules such as requiring that the x if y else z construct be split across three lines.

I separated it into two lines, if you prefer that. I can make one line too. Tell me what's best for you.

Reply:
We can probably just calculate the value once at the top of the function, since we already know the comp_class and whether it is invoked_directly.

Reply:
done